src/Components/ProductPreviewModal/ProductPreviewModal.tsx

```diff
@@ -13,14 +13,34 @@ const ProductPreviewModal = ({productList, onClose}: ProductPreviewModalProps) =
    
   const [activeTab, setActiveTab] = useState('dimensions');
 
+  useEffect(() => {
+    const handleKeyDown = (event) => {
+        if (event.key === 'Escape') {
+            onClose();
+        }
+    };
+
+    document.addEventListener('keydown', handleKeyDown);
```

Comment:
Maybe we can skip this part since the bug is fixed by moving the modal out of ProductCard component wrapper.